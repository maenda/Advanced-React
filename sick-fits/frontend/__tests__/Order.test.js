import Order, { SINGLE_ORDER_QUERY } from '../components/Order';
import { mount } from 'enzyme';
import wait from 'waait';
// import Item from '../components/styles/ItemStyles';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import { fakeOrder } from '../lib/testUtils';

const mocks = [
  {
    // when someone makes a request with this query and variable combo
    request: { query: SINGLE_ORDER_QUERY, variables: { id: 'ord123' } },
    // return this fake data (mocked data)
    result: {
      data: {
        order: fakeOrder()
      }
    }
  }
];

describe('<Order />', () => {
  it('renders with proper data', async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <Order id="ord123" />
      </MockedProvider>
    );

    await wait();
    wrapper.update();

    expect(toJSON(wrapper.find('[data-test="order-id"]'))).toMatchSnapshot();
    expect(toJSON(wrapper.find('[data-test="order-charge"]'))).toMatchSnapshot();
    expect(toJSON(wrapper.find('[data-test="order-date"]'))).toMatchSnapshot();
    expect(toJSON(wrapper.find('[data-test="order-total"]'))).toMatchSnapshot();
    expect(toJSON(wrapper.find('[data-test="order-count"]'))).toMatchSnapshot();
    expect(toJSON(wrapper.find('[data-test="order-items"]'))).toMatchSnapshot();
    expect(wrapper.find('[data-test="order-items"]')).toHaveLength(1);
  });
});
