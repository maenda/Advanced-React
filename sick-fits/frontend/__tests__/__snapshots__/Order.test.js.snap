// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Order /> renders with proper data 1`] = `
<span
  data-test="order-id"
>
  ord123
</span>
`;

exports[`<Order /> renders with proper data 2`] = `
<span
  data-test="order-charge"
>
  ch_123
</span>
`;

exports[`<Order /> renders with proper data 3`] = `
<span
  data-test="order-date"
>
  March, 31, 2018 8:00 PM
</span>
`;

exports[`<Order /> renders with proper data 4`] = `
<span
  data-test="order-total"
>
  $400
</span>
`;

exports[`<Order /> renders with proper data 5`] = `
<span
  data-test="order-count"
>
  2
</span>
`;

exports[`<Order /> renders with proper data 6`] = `
<div
  className="items"
  data-test="order-items"
>
  <div
    className="order-item"
    key="3a430a73-c9e9-4bcc-b46a-41618965ffea"
  >
    <img
      alt="soluta non omnis consequatur enim quia autem"
      src="reprehenderit.jpg"
    />
    <div
      className="item-details"
    >
      <h2>
        soluta non omnis consequatur enim quia autem
      </h2>
      <p>
        Qty: 
        1
      </p>
      <p>
        Each: 
        $42.34
      </p>
      <p>
        Subtotal: 
        $42.34
      </p>
      <p>
        Description: 
        et modi tenetur amet modi reprehenderit omnis
      </p>
    </div>
  </div>
  <div
    className="order-item"
    key="2393e090-592f-4d03-b808-c9d81098deec"
  >
    <img
      alt="quia sed exercitationem omnis laborum exercitationem est"
      src="sint.jpg"
    />
    <div
      className="item-details"
    >
      <h2>
        quia sed exercitationem omnis laborum exercitationem est
      </h2>
      <p>
        Qty: 
        1
      </p>
      <p>
        Each: 
        $42.34
      </p>
      <p>
        Subtotal: 
        $42.34
      </p>
      <p>
        Description: 
        sapiente laudantium molestias assumenda quasi adipisci mollitia
      </p>
    </div>
  </div>
</div>
`;
